### PR TITLE
added DESTROY_OBJECT and DECREMENT_STACKSIZE hooks to Events/ItemEvents

### PR DESCRIPTION
**Pull request by henningkoehlernz**
Can e.g. be used to create infinite ammunition:

on module load:
```
#include "nwnx_events"
void main()
{
    NWNX_Events_SubscribeEvent("NWNX_ON_ITEM_DECREMENT_STACKSIZE_BEFORE", "on_decr_stack");
}
```
on_decr_stack:
```
#include "nwnx_events"
void main()
{
   if ( IsMarkedAsInfinite(OBJECT_SELF) )
       NWNX_Events_SkipEvent();
}
```

### CODE REVIEW

**Comment by Daztek** [comment]
Might be good to mention that this conflicts with the (Item)ObjectEventHandler target of NWNX_Profiler. ObjectEventHandler profiling is off by default so it's not a super big deal, but adding a note is probably good.

File: CHANGELOG.md [hunk 1]
```diff
@@ -110,6 +110,7 @@ The following plugins were added:
 - Encounter: GetResetTime()
 - Encounter: SetResetTime()
 - Events: GetCurrentEvent()
+- Events: ON_ITEM_EVENT_{BEFORE|AFTER}
```
Comment (Daztek) [inline CHANGELOG.md]: Minor, this should be added to `Events: Added ItemInventory{Open/Close}, ItemInventory{Add/Remove}Item and ScrollLearn events to ItemEvents` under the `Added` header in the changelog.

File: Plugins/Events/NWScript/nwnx_events.nss [hunk 2]
```diff
@@ -729,6 +739,11 @@ const int NWNX_EVENTS_TIMING_BAR_LOCK          = 8;
 const int NWNX_EVENTS_TIMING_BAR_CUSTOM        = 10;
 */
 
+/*
+const int NWNX_EVENTS_ITEM_EVENT_DESTROY_OBJECT = 11;
+const int NWNX_EVENTS_ITEM_EVENT_DECREMENT_STACKSIZE = 16;
+*/
```
Comment (Daztek) [inline Plugins/Events/NWScript/nwnx_events.nss]: Are these the only events Items have?

Comment (henningkoehlernz) [reply Plugins/Events/NWScript/nwnx_events.nss]: These are the only ones I've managed to trigger. The 1.69 decompiled code also handles EVENT_TIMED_EVENT, EVENT_APPLY_EFFECT and EVENT_REMOVE_EFFECT, but at least the latter two are not handled anymore (signature changed and ApplyEffectToObject doesn't trigger anything).

**Pull request merged**